projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/asm/LLVMAMD64CpuidNode.java

```diff
@@ -29,20 +29,60 @@
  */
 package com.oracle.truffle.llvm.nodes.asm;
 
+import com.oracle.truffle.api.CompilerAsserts;
 import com.oracle.truffle.api.dsl.NodeChild;
 import com.oracle.truffle.api.dsl.Specialization;
 import com.oracle.truffle.api.frame.VirtualFrame;
-import com.oracle.truffle.llvm.nodes.asm.support.LLVMAMD64WriteRegisterNode.LLVMAMD64WriteI32RegisterNode;
+import com.oracle.truffle.llvm.nodes.asm.support.LLVMAMD64WriteValueNode;
 import com.oracle.truffle.llvm.runtime.nodes.api.LLVMExpressionNode;
 
 @NodeChild("level")
 public abstract class LLVMAMD64CpuidNode extends LLVMExpressionNode {
-    @Child private LLVMAMD64WriteI32RegisterNode eax;
-    @Child private LLVMAMD64WriteI32RegisterNode ebx;
-    @Child private LLVMAMD64WriteI32RegisterNode ecx;
-    @Child private LLVMAMD64WriteI32RegisterNode edx;
+    public static final String BRAND = "Sulong"; // at most 48 characters
+    public static final String VENDOR_ID = "SulongLLVM64"; // exactly 12 characters
 
-    public LLVMAMD64CpuidNode(LLVMAMD64WriteI32RegisterNode eax, LLVMAMD64WriteI32RegisterNode ebx, LLVMAMD64WriteI32RegisterNode ecx, LLVMAMD64WriteI32RegisterNode edx) {
+    public static final int[] BRAND_I32 = getI32(BRAND, 12);
```

Comment:
You can add `@CompilationFinal(dimensions = 1)` to these two arrays. Do the fields need to be public?